File: ruby/04-cryptic-calculator/src/calculator.rb
# frozen_string_literal: true

class Calculator
  def normalize(a, b, c)
    d = []
    e = Float::INFINITY
    f = -Float::INFINITY
    a.each do |x|
      e = x if x < e
      f = x if x > f
    end
    h = f - e
    i = c - b
    a.each do |x|
      k = b + ((x - e).to_f / h) * i
      d << k.round(2)
    end
    d
  end
end

File: javascript/04-cryptic-calculator/src/calculator.js
class Calculator {
  constructor() {}

  normalize(a, b, c) {
    let d = [];
    let e = Infinity;
    let f = -Infinity;

    for (let g = 0; g < a.length; g++) {
      if (a[g] < e) e = a[g];
      if (a[g] > f) f = a[g];
    }

    let h = f - e;
    let i = c - b;

    for (let g = 0; g < a.length; g++) {
      let j = (a[g] - e) / h;
      let k = b + j * i;
      d.push(Math.round(k * 100) / 100);
    }

    return d;
  }
}

export { Calculator };
